### DISCUSSION TITLE
Make `filepath` on `NodeOnDiskFile` public

### DISCUSSION THREAD

**Discussion by cstrnt** [remix-run/remix#3021] [Proposals]
I'm currently fiddling around with Remix and while using `unstable_createFileUploadHandler` I've noticed that the `filePath` property on a  `NodeOnDiskFile` is private. I don't think that this is really useful because I want to do some cleanup after using the file.

If it's better to keep the `filePath` property private, I think it would be nice to add a function called `remove` or `cleanup` that unlinks the file in the filesystem. 

I know that it's marked as unstable, but multipart uploads are quite important for many applications :)

**Comment by vitormv** [comment]
I also have to use a pdf parsing library that does not accept a file handle, but rather a string for a file path. And in this current way its implemented, I would need to create another temp file just to be able to pass that string to the library. Would be much easier to just have access to where the original file is saved.

**Comment by machour** [comment]
There's an open PR for adding `getFilePath()` and `remove()` to `NodeOnDiskFile`: https://github.com/remix-run/remix/pull/4408

**Comment by julienmonnard** [comment]
Although not in the changelogs, #4408  is now merged and part of v1.12.0!
